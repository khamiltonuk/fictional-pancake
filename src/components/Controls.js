import React from "react";

const Controls = props => {
  const operations = ["rock", "paper", "sissor", "spock", "lizard"];
  return (
    <nav class="actions">
      {operations.map((e, i) => (
        <button data-q={`make-move-${e}`} key={i} onClick={() => makeMove(e)}>
          {e}
        </button>
      ))}
    </nav>
  );
};

export default Controls;
