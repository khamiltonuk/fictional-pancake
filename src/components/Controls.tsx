import React, { ReactElement, Dispatch } from "react";
import styled from "styled-components";
import { ActionT, makeMove, animateHands } from "../actions";

const Action = styled.nav`
  position: absolute;
  bottom: 10px;
  text-align: center;
  width: 100%;
  transition: bottom 0.2s ease;
  &.disabled {
    bottom: -100%;
  }
`;

const Button = styled.button`
  text-transform: uppercase;
  display: inline-block;
  background: #da727e;
  padding: 10px 20px;
  border: none;
  color: #000;
  position: relative;
  width: 100%;
  position: static;
  margin-bottom: 5px;
  &:focus,
  &:hover {
    background: #ac6c82;
  }
  &:disabled {
    opacity: 0.3;
  }
  @media (min-width: 640px) {
    margin: 0;
    width: auto;
  }
`;

type PropsT = {
  moves: string[];
  dispatch: Dispatch<ActionT>;
  isGameOver: string;
};

const Controls: React.FunctionComponent<PropsT> = ({
  moves,
  dispatch,
  isGameOver,
}: PropsT): ReactElement => {
  function handleMove(move: string) {
    dispatch(animateHands());

    setTimeout(() => dispatch(makeMove(move)), 1000);
  }

  return (
    <Action>
      {moves.map((move) => (
        <Button
          data-q={`make-move-${move}`}
          key={move}
          onClick={() => handleMove(move)}
          disabled={isGameOver !== ""}
        >
          {move}
        </Button>
      ))}
    </Action>
  );
};
export default Controls;
