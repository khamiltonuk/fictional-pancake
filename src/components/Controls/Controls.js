import React from "react";
import styled from "styled-components";

const Action = styled.nav`
  position: absolute;
  bottom: 10px;
  text-align: center;
  width: 100%;
  transition: bottom 0.2s ease;
  &.disabled {
    bottom: -100%;
  }
`;

const Button = styled.button`
  text-transform: uppercase;
  display: inline-block;
  background: #da727e;
  padding: 10px 20px;
  border: none;
  color: #000;
  margin: 0;
  position: relative;
`;

const Controls = props => {
  const makeMove = () => {
    console.log("hello");
  };
  return (
    <Action>
      {props.moves.map((e, i) => (
        <Button data-q={`make-move-${e}`} key={i} onClick={() => makeMove(e)}>
          {e}
        </Button>
      ))}
    </Action>
  );
};

export default Controls;
