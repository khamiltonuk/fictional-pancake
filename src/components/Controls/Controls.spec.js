import React from "react";
import { shallow } from "enzyme";
import Controls from "./Controls";
import toJson from "enzyme-to-json";

describe("Controls", () => {
  test("should match the snapshot", () => {
    const props = ["rock", "paper", "cheese"];
    const wrapper = shallow(<Controls moves={props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should render 3 buttons if given 3 moves", () => {
    const props = ["rock", "paper", "cheese"];
    const wrapper = shallow(<Controls moves={props} />);
    expect(wrapper.find('[data-q*="make-move"]').length).toBe(3);
  });
});
