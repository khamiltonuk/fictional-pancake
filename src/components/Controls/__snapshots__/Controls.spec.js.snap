// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match the snapshot 1`] = `
<styled.nav>
  <styled.button
    data-q="make-move-rock"
    key="0"
    onClick={[Function]}
  >
    rock
  </styled.button>
  <styled.button
    data-q="make-move-paper"
    key="1"
    onClick={[Function]}
  >
    paper
  </styled.button>
  <styled.button
    data-q="make-move-cheese"
    key="2"
    onClick={[Function]}
  >
    cheese
  </styled.button>
</styled.nav>
`;
